### ISSUE TITLE
Removing the NavBar causes a white field to appear in its place

### ISSUE THREAD

**Issue by jkhoel** [dotnet/maui#11611]
### Description

The page is a `ContentPage` with a `Grid`. I have set `Shell.BackgroundColor` and I have set `Shell.NavBarIsVisible="False"`. On iOS, this leaves a white field in the place where the NavBar used to be:
![Simulator Screen Shot - iPhone 14 Pro - 2022-11-24 at 15 16 04](https://user-images.githubusercontent.com/13721596/203806616-9efe71dd-3969-4ff9-b9a1-da44cce4dbc3.png)

If I enable the navbar, the white field goes away - but then I get the text I don't want to see instead:
![Simulator Screen Shot - iPhone 14 Pro - 2022-11-24 at 15 16 22](https://user-images.githubusercontent.com/13721596/203806718-f9d21922-647e-4761-a732-885203bbc849.png)

### Steps to Reproduce

1. Create a new MAUI project
2. Create a ContentPage with a Grid
3. Set `<ContentPage Shell.NavBarIsVisible="False" Shell.BackgroundColor="hotpink" ...other default options/>`
4. Target iOS

### Link to public reproduction project repository

-

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS latest

#### Workaround 1

Enable `Shell.NavBarIsVisible` and set the page as having no title

#### Workaround 2

Disable `SafeArea` for iOS devices for particular pages by using platform specific services:
```xaml
<ContentPage
     ... whatever else you might have here
     xmlns:ios="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls"
     ios:Page.UseSafeArea="False" >
```

**Comment by jsuarezruiz** [comment]
Happens on iOS16 and with dynamic island.

**Comment by jkhoel** [comment]
Updated with two possible workarounds. This seem to be related to `SafeAreaView`s

**Comment by Ghevi** [comment]
This happens in Maui Blazor too in iOS devices like IPhone 14 pro Max, doesn't happen in an phisical IPhone SE which has a smaller screen. 
I tried inspecting the `BlazorWebView ` with Safari inspector and the small white gap at the top is not part of it, so it must be the `AppShell`.
I tried using a NavigationPage instead of the shell, but I get an error `InvalidOperationException: Android activity cannot be null` when instanciating the `MainPage` on the` InitializeComponent()`

Alsoo, maybe I'm doing something wrong with the css, but if a remove the safearea on iOS the app goes below the status bar, like it's very pushed to the top.
